Kulaev Evgeny. Laboratory work №1. Option №2

zcotti opened the pull request:
A plugin that adds functions without conditions to the `always_inline` attribute and outputs it

zcotti committed:
first commit

aobolensk commented on clang/test/lab1/kulaev_zhenya/test.cpp:
```diff
@@ -0,0 +1,18 @@
+// RUN: %clang_cc1 -load %llvmshlibdir/AttrubuteAlwaysPlugin%pluginext -plugin always_inlines-plugin %s 1>&1 | FileCheck %s
+
+// CHECK: __attribute__((always_inline)) sum
```
Please, check that attribute actually appeared

aobolensk commented on clang/labs/lab1/kulaev_zhenya/AttrubuteAlways.cpp:
```diff
@@ -0,0 +1,58 @@
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+
+using namespace clang;
```
Please, use clang namespace explicitly, avoid using "using namespace *"

aobolensk commented:
Please, update the branch to go through updated clang-format check

m-ly4 commented on clang/labs/lab1/kulaev_zhenya/AttrubuteAlways.cpp:
```diff
@@ -0,0 +1,58 @@
+    return std::make_unique<AlwaysInlineConsumer>(&Compiler.getASTContext());
+  }
+  bool ParseArgs(const CompilerInstance &Compiler,
+                 const std::vector<std::string> &args) override {
```
Could you please fix warnings printed by clang-tidy test: https://github.com/NN-complr-tech/llvm-nnsu-2024/actions/runs/8250927820/job/22566738976?pr=6

m-ly4 replied:
There are still some warnings in clang-tidy that can be fixed.

zcotti replied:
We will fix it)

zcotti committed:
Merge branch 'NN-complr-tech:course-spring-2024' into kulaev_e_var2

zcotti committed:
fix errors

zcotti commented:
1) Fixed previous errors
2) Fixed clang-format
3) Fixed a bug with clang-tidy

@m-ly4  But my windows crashed, and apparently not because of me, Polozov Vladislav had a similar error, how can I fix it?

zcotti committed:
fix clang-tidy

m-ly4 committed:
Merge branch 'course-spring-2024' into kulaev_e_var2

aobolensk commented on clang/test/lab1/kulaev_zhenya/test.cpp:
```diff
@@ -0,0 +1,31 @@
+  }
+}
+
+// CHECK: Added attribute always_inline in checkEmpty
```
Please add check for the attribute, not for logs that you print by yourself

m-ly4 committed:
Merge branch 'course-spring-2024' into kulaev_e_var2

zcotti committed:
Merge branch 'course-spring-2024' into kulaev_e_var2

zcotti committed:
fix test

m-ly4 committed:
Merge branch 'course-spring-2024' into kulaev_e_var2
